src/imp/linux_raw/process/syscalls.rs

```diff
@@ -387,6 +390,97 @@ pub(crate) fn getrlimit(limit: Resource) -> Rlimit {
     }
 }
 
+#[inline]
+pub(crate) fn setrlimit(limit: Resource, new: Rlimit) -> io::Result<()> {
+    let lim = linux_raw_sys::v5_4::general::rlimit64 {
+        rlim_cur: new
+            .current
+            .unwrap_or(linux_raw_sys::general::RLIM_INFINITY as _),
+        rlim_max: new
+            .maximum
+            .unwrap_or(linux_raw_sys::general::RLIM_INFINITY as _),
+    };
+    #[cfg(target_pointer_width = "32")]
+    unsafe {
+        match ret(syscall4(
```

Comment:
I only skimmed this code, seems OK.  I just have to express my reservations on an ongoing basis around the direct syscall path.  There are so many corner cases and bits of complexity to consider around 64 vs 32 bit, any architecture oddities etc.

(But again, the code in a quick skim seems fine)

Reply:
Acknowledged. I think Rust's general strictness about integer conversions, along with `try_into()` and `Result` are great tools that help a lot here, compared to my experience writing this kind of code in C. But even so, bugs in this area in practice are something I'm keeping an eye out for, because above all, rustix needs to be a practical library.